src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1678,7 +1682,14 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
         {
             assert(!isRMW);
 
-            srcCount += BuildOperandUses(intrin.op2, RBM_ASIMD_INDEXED_H_ELEMENT_ALLOWED_REGS.GetFloatRegSet());
+            if (intrin.id == NI_Sve_DuplicateSelectedScalarToVector)
```

Comment:
hhm, in both cases, we were using float register candidates and with your fix, we are just using reduced float register candidates. how was that causing an issue where we were just associating wrong type of candidates to a node?

Reply:
Or was it the case, where type of candidates were correct, but the actual set should have been different, which you are handling it here?